security/fma-shared-lockbox.md

```diff
@@ -0,0 +1,147 @@
+
+- **Description:** The `clusterManager` role in `SuperchainConfigInterop` can directly call `addDependency` on L1, bypassing the normal L2 → L1 flow. If compromised, an attacker could add malicious or arbitrary chain IDs and authorize their own portal.
+- **Risk Assessment:** High.
+    - Potential Impact: Critical. A malicious addition to the dependency set could allow an attacker to `unlockETH` from the `SharedLockbox`.
+    - Likelihood: Low. The `clusterManager` should be a highly secured role (e.g., a multi-sig or governance key).
+- **Mitigations:** Strictly secure the `clusterManager` key.
+- **Detection:** Track the `DependencyAdded` event for calls from `clusterManager`. Compare newly added chain IDs to an approved list.
+- **Recovery Path(s):** Pause the system through `SuperchainConfig`. Since removal functions are unavailable, an upgrade directly over **`SuperchainConfigInterop`** to clear storage slots will be required.
+
+### FM3: Unauthorized or incorrectly initialized `superchainConfig` address in `OptimismPortalInterop`.
+
+- **Description:** In both `SharedLockbox` and `OptimismPortalInterop`, the address of the `SuperchainConfig` is stored in a special storage slot. If it’s set incorrectly or to a malicious address, the entire access-control scheme breaks.
+- **Risk Assessment:** High.
+    - Potential Impact: Critical. Since the entire access-control scheme depends on `SuperchainConfig`, all funds could potentially be extracted from the **OptimismPortal**.
+    - Likelihood: Low. Storage slots and values are carefully set during protocol upgrades.
+- **Mitigations:** There should be tests for the initialization step, and ensuring that `superchainConfig` is only set once.
+- **Detection:** Verify the final `superchainConfig` address in newly deployed or upgraded contracts.
+- **Recovery Path(s):** Pause the system, to upgrade the contract to set `superchainConfig` correctly.
+
+### FM4: Solidity Compiler generates an exploitable Bytecode
+
+- **Description:** Smart contracts are typically audited based on their **source code**, assuming that the Solidity compiler (`solc`) produces a correct and verifiable output. Relying solely on source code audits, without verifying the actual bytecode, could introduce undetected vulnerabilities.
+- **Risk Assessment:** High.
+    - Potential Impact: Critical. If the Solidity compiler generates incorrect or exploitable bytecode, it could introduce a hidden attack vector.
+    - Likelihood: Low. The Solidity compiler is widely used and battle-tested, but compiler bugs have occurred in the past.
+- **Mitigations:** Use a widely audited Solidity version. ****Compare the Keccak-256 hash of the compiled bytecode against an expected reference. Track Solidity compiler CVEs to be aware of past and ongoing issues.
+- **Detection:** Verify deployed bytecode against expected hash values before proxy to target the implementation contract.
```

Comment:
Staying on top of new solidity bugs is the other detection here. Though in reality if something bad happened, solidity team would likely reach out to us as part of incident response efforts